### DISCUSSION TITLE
"delay" in rules skipped

### DISCUSSION THREAD

**Discussion by tsunfinger** [arendst/Tasmota#19519] [Q&A]
I have two rules running. rule1 is using "delay" command.
When rule2 is triggered while rule1 is running, the "delay" command in rule 1 will be skipped.
Is this the expected behaviour?

**Comment by sfromis** [comment]
Do not use `Delay` if you expect reliable timing, it can be cut short by other things happening. Use a `RuleTimer`

**Comment by tsunfinger** [comment]
Thanks for fast reply.
I controlling an H-bridge. Switching on/off is controlled by same rule. It's very important for the time "on" is executed a switch to "off" will be ignored. This is done by disabling rule while being executed.
Will "RuleTimer" work on a rule which is temporarily disabled? I probably could split into different rules, but only 3 rules are supported.

rule1
on Power1#state=0 do 
backlog rule1 0; PWM8 1023; PWM9 480; delay 385; PWM9 400; delay 7; PWM9 1023; rule1 1
endon
on Power1#state=1 do
backlog rule1 0; PWM9 1023; PWM8 480; delay 385; PWM8 400; delay 7; PWM8 1023; rule1 1
endon

**Reply by sfromis** [reply]
Your rules and Delay are really bad. :grinning: 

RuleTimer working is not related to disabling a rule set or not, but each delay is better done with a RuleTimer, and the rule to handle the RuleTimer should not be part of the rule set you disable/enable.

Instead of at all disabling/enabling rule set, I'd use another strategy, like `on Power1#state=%var3%` and then set var3 depending on when you want the trigger to match or not.

However, since you seem to be using ESP32, what I'd really prefer would be to just not mess with rules at all, but instead manage the actions using the "regular programming language" of Berry.

**Comment by tsunfinger** [comment]
Thanks for hint with variable. Seems to be working with all requirements inside one rule
